src/sitetype.jl

```diff
@@ -12,7 +12,7 @@ end
 function siteinds(sitetypes::AbstractDictionary, g::AbstractGraph; kwargs...)
   is = IndsNetwork(g)
   for v in vertices(g)
-    is[v] = [siteind(sitetypes[v], v; kwargs...)]
+    is[v] = [siteind(sitetypes[v], vertex_tag(v); kwargs...)]
```

Comment:
I see, so for a case like this:
```julia
using Graphs
using NamedGraphs
using ITensors
using ITensorNetworks

g = NamedGraph()
add_vertex!(g, "(1,2)")
s = siteinds("S=1/2", g)
```
it will return:
```julia
IndsNetwork{Any, Index} with 1 vertices:
1-element Vector{Any}:
 "(1,2)"

and 0 edge(s):

with vertex data:
1-element Dictionaries.Dictionary{Any, Vector{Index}}
 "(1,2)" │ Index[(dim=2|id=620|"S=1/2,Site,n=1×2")]

and edge data:
0-element Dictionaries.Dictionary{NamedEdge{Any}, Vector{Index}}
```
instead of:
```julia
IndsNetwork{Any, Index} with 1 vertices:
1-element Vector{Any}:
 "(1,2)"

and 0 edge(s):

with vertex data:
1-element Dictionaries.Dictionary{Any, Vector{Index}}
 "(1,2)" │ Index[(dim=2|id=757|"2),S=1/2,Site,n=(1")]

and edge data:
0-element Dictionaries.Dictionary{NamedEdge{Any}, Vector{Index}}
```
so it is useful but I guess causes some other problem.

Reply:
Looks like we can keep this line as-is but change line 6 of this file to:
```julia
function siteind(d::Integer, v; addtags="", kwargs...)
```
(i.e. remove the type constraint on `v`).